src/compose/ComposeMenu.js

```diff
@@ -234,56 +227,58 @@ class ComposeMenuInner extends PureComponent<Props> {
   });
 
   render() {
-    const { expanded, insertVideoCallLink, onExpandContract } = this.props;
-    const numIcons =
-      2 + (Platform.OS === 'android' ? 1 : 0) + (insertVideoCallLink !== null ? 1 : 0);
+    const { expanded, insertVideoCallLink } = this.props;
+
+    const onOpenActionSheet = () => {
+      const options = [
+        'Upload a file',
+        'Upload a picture',
+        'Take a picture',
+        'Add a video call',
+        'Cancel',
+      ];
+      const cancelButtonIndex = options.length - 1;
+
+      this.props.showActionSheetWithOptions(
+        {
+          options,
+          cancelButtonIndex,
+        },
+        buttonIndex => {
+          if (buttonIndex === 0) {
+            this.handleFilesPicker();
+          }
+          if (buttonIndex === 1) {
+            this.handleImagePicker();
+          }
+          if (buttonIndex === 2) {
+            this.handleCameraCapture();
+          }
+          if (buttonIndex === 3) {
+            // Flow actually has a special case for this, this is called an "unknown function".
+            // We know that insertVideoCallLink is a function, but we don't know anything about its
+            // arguments or return type so we can't safely do anything with it except pass it around.
+            // see:https://stackoverflow.com/questions/65314559/flow-type-refine-mixed-to-function
+            if (typeof insertVideoCallLink === 'function') {
+              insertVideoCallLink();
```

Comment:
> we don't know anything about its arguments or return type so we can't safely do anything with it except pass it around

That's not true 🙂—we're not just passing this function around. We're calling it, and Flow doesn't complain. We're allowed to do so because (unlike the case in that SO thread) Flow knows specifically what the function type will be:

```js
type OuterProps = $ReadOnly<{|
  // …
  insertVideoCallLink: (() => void) | null,
  // …
|}>;
```

i.e., it takes no arguments and it returns `undefined`.

In any case, there's a bug here: we shouldn't offer the action-sheet button in the first place if nothing happens when you press it.